### ISSUE TITLE
Incorrect warning "the trait .. is not implemented for .."

### ISSUE THREAD

**Issue by nanoqsh** [intellij-rust/intellij-rust#7685]
## Environment

* **IntelliJ Rust plugin version: 0.4.152.4026**
* **Rust toolchain version: stable-x86_64-unknown-linux-gnu**
* **IDE name and version:  Intellij IDEA 2021.1.3 (Community Edition)**
* **Operating system: Arch Linux**

## Problem description
Incorrect IDE warning:
```
the trait bound .. is not satisfied [E0277]
the trait .. is not implemented for ..
```
Although all requirements are satisfied and the code compiles successfully without errors.

## Steps to reproduce
```rust
#[derive(Debug)]
struct Content<T, const N: usize>([T; N]);

impl<T: PartialEq, const N: usize> PartialEq for Content<T, N> {
    fn eq(&self, other: &Self) -> bool { self.0 == other.0 }
}

impl<T: Eq, const N: usize> Eq for Content<T, N> {}
//                          ^^
// the trait bound `Content<T, { N }>: PartialEq<Content<T, { N }>>` is not satisfied [E0277]
// the trait `PartialEq<Content<T, { N }>>` is not implemented for `Content<T, { N }>`

fn main() {
    let a = Content([1, 2]);
    let b = Content([1, 2]);
    assert_eq!(a, b)
}
```
